### ISSUE TITLE
setDataSource only accepts absolute urls

### ISSUE THREAD

**Issue by fermuch** [tangrams/tangram#591]
**TO REPRODUCE THE ISSUE, FOLLOW THESE STEPS:**

- Create a basic map
- Update a source with `setDataSource`, using a relative url (`/data.json` instead of `http://example.com/data.json`)

**RESULT:**

You get the following error in the console:

```
data source load error(s) for source 'data', URL '/data.json': Error: Failed to execute 'open' on 'XMLHttpRequest': Invalid URL
```

**EXPECTED RESULT:**

Relative urls should work as they do in the tangram YAML definition.
The base url of the current scene should be used as base path.

**ENVIRONMENT:**

- Browser: Opera Developer, and Chrome.
- OS: Linux 64bits

**OTHER:**

The expected result is just a recommendation. I'm not sure if using the base path is correct in this context.

**Comment by bcamper** [comment]
The expected behavior here is that for a scene loaded from `http://example.com/base/scene.yaml`:

- A *path-relative* URL such as `path/to/data.json` would resolve relative to the *scene file path*:
  - `http://example.com/base/path/to/data.json`
- A *domain-relative* URL such as `/data.json` would resolve relative to the scene's *domain* (regardless of the path of the scene file):
  - `http://example.com/data.json`

That appears to be the behavior I'm seeing with `setDataSource()`, do you have an example that is not following that?

**Comment by fermuch** [comment]
Of course!

Take a look here: https://fermuch.ml/tangram-demo/
**Expected result:** to see a red and a blue point.
**Actual result:** Only a red point (loaded from sources) is shown. The blue point, loaded from a relative url with `setDataSource`, throws a warning and doesn't load.

----

![red-dot](https://user-images.githubusercontent.com/998598/30329742-8d99084c-97a9-11e7-9883-c19e1816d825.png)
The red dot.

![blue-dot-error](https://user-images.githubusercontent.com/998598/30329764-9f5700f2-97a9-11e7-9065-b6d222ef62ae.png)
The error 

-----

There are some comments on index.html and scene.yaml to mark where things are loaded and how it is not loading correctly.

**Comment by meetar** [comment]
Hi @fermuch – unfortunately that site seems to be down, I've checked it a few times over the last week and I always see "fermuch.ml took too long to respond. ERR_CONNECTION_TIMED_OUT". Are the files available anywhere else?

**Comment by fermuch** [comment]
The domain just expired! Here's a zip of the files:

[tangram-demo.zip](https://github.com/tangrams/tangram/files/1321638/tangram-demo.zip)

**Comment by meetar** [comment]
Hi @fermuch – I see you're using Tangram v 0.10, this issue was fixed in version 0.11.7: https://github.com/tangrams/tangram/releases/tag/v0.11.7

We encourage you to use the latest release of Tangram, you can always get the current version by using this URL: https://mapzen.com/tangram/tangram.min.js

Also, rather than setting a timeout you can bind to Tangram's `load` event like so:
```
layer.scene.subscribe({
  load: function (e) {
    layer.scene.setDataSource('test1', {
      type: 'GeoJSON',
      url: 'demo.json'
    });
  }
});
```

**Comment by fermuch** [comment]
:+1: Thanks
